websrc/gadget_menu/src/gadget_menu_main.tsx

```diff
@@ -306,19 +340,65 @@ class ControlPanel extends React.Component< {}, ControlPanelState >
 			{
 				this.updateSettings();
 			}
-		} );
+		} )
+		.catch( (reason: any) => 
+		{
+			console.log( `Get installed gadgets failed with ${ JSON.stringify( reason ) }` );
+		});
 
 		for( let favorite of this.settings.favorites )
 		{
-			this.requestManifest( favorite );
+			this.requestManifest( favorite )
+			.catch( (reason: any) =>
+			{
+				console.log( `Get gadget manifest for ${ favorite } failed with ${ JSON.stringify( reason ) }` );
+			});
 		}
 
 		for( let builtin of k_alwaysInstalledGadgets )
 		{
-			this.requestManifest( builtin );
+			this.requestManifest( builtin )
+			.catch( (reason: any) =>
+			{
+				console.log( `Get gadget manifest for ${ builtin } failed with ${ JSON.stringify( reason ) }` );
+			});
 		}
 
-		this.requestManifest( k_desktopWindowGadget );
+		this.requestManifest( k_desktopWindowGadget )
+		.catch( (reason: any) =>
+		{
+			console.log( `Get gadget manifest for desktop window gadget failed with ${ JSON.stringify( reason ) }` );
+		});
+
+		this.autoLaunchGadgets()
+		.catch( (reason: any) =>
+		{
+			console.log( `Auto launch gadgets failed: ${ JSON.stringify( reason ) }` );
+		});
+	}
+
+	@bind
+	private autoLaunchGadgets()
+	{
+		return new Promise<any>( async ( resolve, reject ) => 
+			{
+				console.log( `starting gadgest ${JSON.stringify( this.settings.autoLaunch) }` )
+				for ( let autoLaunch of this.settings.autoLaunch )
+				{
+					let res = await AvGadget.instance().startGadget( autoLaunch, [] )
+					if (res.success)
+					{
+						console.log( `started gadget ${ autoLaunch }` );
+						resolve();
+					} 
+					else
+					{
+						console.log( `gadget start failed ${ autoLaunch }` );
+						reject();
```

Comment:
Won't this kill any autolaunch gadgets after the one that failed? Maybe that's not so good?

This loop will also wait for one gadget to load before loading the next. Seems like they should be allowed to start in parallel so total wall clock time for the user is minimized.

You can build an array of startGadget promises and then I think Promise has a "wait on this array of promises" method.

Reply:
switched it to promise.all but I think that may early out as well.

Reply:
It will likely early out if one fails, but only after all the requests have been made. So it will start all the gadgets, just return from this function a bit early.

Reply:
will do some extra testing around this also